cobalt/loader/resource_cache.h

```diff
@@ -347,6 +349,7 @@ void CachedResource<CacheType>::OnContentProduced(
   DCHECK_CALLED_ON_VALID_THREAD(cached_resource_thread_checker_);
   DCHECK(!resource_);
 
+  if (is_concealed_) return;
```

Comment:
If you have `FinishInternal`  signal an error by returning nullptr for the image, knowing about `is_concealed_` here shouldn't be necessary because the `image_available_callback_` callback that leads here won't be called anymore (See `ImageDecoder::Finish`).

Reply:
I'm not sure how `FinishInternal` would know to signal an error—the order of events is such that it doesn't yet know that the Conceal has happened, so we have to do it here instead

Reply:
It can know by checking the boolean set synchronously when the conceal is received...

Reply:
Are you talking about the boolean `is_concealed_` that's also being added in this PR? One race condition that can happen is `FinishInternal` gets called before `is_concealed_` is set, in which case it will end up adding the image loaded callback (this function) to the queue of MainWebModule tasks. So the idea here is to have knowledge of the conceal on the MainWebModule thread before running the image loaded callback.

Reply:
Since `FinishInternal` runs on the WebModule thread, as does `WebModule::Impl::Reveal` that calls `ImageCache::OnReveal()` in this PR and `ResourceCache<CacheType>::ReclaimMemory()` that calls `it.second->SetIsConcealed` in this PR, so that is the same thread, can only happen in sequence so there can not be a race.

Reply:
Knowing about `concealed` state here doesn't obviously seem to make anything worse off, so i think it's okay to have explicit signaling here.

Then we don't rely on behavior of a number of other modules that may or may not retain the same call order of processing order - it makes refactorings easier. 

If the thesis is that "this will never run here because of other code" i'd add a `NOTREACHED` to assert that before doing a `return;`